### PR TITLE
Add stable diffusion XL tests and fixes batch size bug

### PR DESCRIPTION
**Pull request by skavulya**
Adds stable diffusion XL tests and fixes bug when generated images not divisible by batch size.

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/stable-diffusion-xl/README.md [hunk 1]
```diff
@@ -83,3 +83,24 @@ python text_to_image_generation.py \
 > The first batch of images entails a performance penalty. All subsequent batches will be generated much faster.
 > You can enable this mode with `--use_hpu_graphs`.
 
+### SDXL-Turbo
+SDXL-Turbo is a distilled version of SDXL 1.0, trained for real-time synthesis. 
+
+Here is how to generate images with multiple prompts:
+```python
```
Comment (dsocek) [inline examples/stable-diffusion-xl/README.md]: python -> bash

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 2]
```diff
@@ -886,7 +886,7 @@ def __call__(
             timestep_cond = None
             if self.unet.config.time_cond_proj_dim is not None:
                 guidance_scale_tensor = torch.tensor(self.guidance_scale - 1).repeat(
-                    batch_size * num_images_per_prompt
+                    num_prompts * num_images_per_prompt
```
Comment (dsocek) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py]: I personally think we should submit this as a separate PR.  I also noted some small fixes we could create for the original SD.  So we can submit this separately as its not part of SDXL.  However, lets not forget, and lets merge with few other smaller fixes (I think there are some in the callbacks) as well as some typos

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_xl.py [hunk 3]
```diff
@@ -1179,15 +1179,14 @@ def __call__(
...
-            #timesteps, num_inference_steps = retrieve_timesteps(self.scheduler, num_inference_steps, device, timesteps)
             self.scheduler.set_timesteps(num_inference_steps, device="cpu")
             timesteps = self.scheduler.timesteps.to(device)
             self.scheduler.reset_timestep_dependent_params()
 
             # 5. Prepare latent variables
             num_channels_latents = self.unet.config.in_channels
             latents = self.prepare_latents(
-                batch_size * num_images_per_prompt,
+                num_prompts * num_images_per_prompt,
```
Comment (dsocek) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_xl.py:1189]: Good catch!

File: tests/test_diffusers.py [hunk 4]
```diff
@@ -730,3 +733,344 @@ def test_no_generation_regression_upscale(self):
...
+        images = sd_pipe(
+            prompt,
+            num_inference_steps=2,
+            output_type="np",
+            batch_size=batch_size,
+            num_images_per_prompt=num_images_per_prompt,
+        ).images
+
+        self.assertEqual(len(images), num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (64, 64, 3))
```
Comment (dsocek) [inline tests/test_diffusers.py:996]: Seems that default images are of size 1024x1024 -->  128x128 latents. (see https://github.com/huggingface/diffusers/blob/main/docs/source/en/using-diffusers/sdxl.md#text-to-image).  Not sure how here we are working with 64x64 here unless we have a bug and SDXL pipe default is 512x512?  Let's check.

I think the recommended square size for XL is 1024x1024 (see https://stability.ai/sdxl-aws-documentation).  512x512 likely to functionally work but could yield bad looking image.   Maybe OK for CI tests.

**Review by dsocek — CHANGES_REQUESTED** [review]
Let's address some of the in-code comments before merging to SDXL branch

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py (hunk 2 above)
Comment (skavulya) [reply optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py]: Agreed. I'll revert the change and we can submit in separate PR.

File: tests/test_diffusers.py (hunk 4 above)
Comment (skavulya) [reply tests/test_diffusers.py:996]: Good point. I copied the test model from diffusers which is using a smaller unet. I think they are using the smaller size for faster tests. https://github.com/huggingface/diffusers/blob/main/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L164

Comment (dsocek) [reply tests/test_diffusers.py:996]: Ok so diffusers is also using wrong unet :)  In that case lets use the same, although not correct.   We could try to submit PR for fix to diffusers (or open a new issue), and if they fix, we can too.  But lets put that on back burner for now.

**Review by dsocek — APPROVED** [review]
LGTM
